### PR TITLE
Generate multiple augmentations from ffcv dataloader

### LINKED ISSUE #31: Incorporate multiple patches in ffcv ssl dataloader
Reinstall ffcv using FFCV-SSL codebase and incorporate functionality to generate multiple patches

- [ ] Reinstall ffcv using FFCV-SSL code base
- [ ] Test generating 2 augmentations using train.beton files instead of current double_train.beton files
- [ ] Update forward pass to adhere to new dataloader yield specs (order of images and labels might be different)
- [ ] Add functionality to generate arbitrary number of patches/augmentations from each image

### PR DESCRIPTION
**Pull request by arnaghosh**
**[WIP] TL;DR: FFCV dataloaders for Cifar-10 changed to generate multiple patches for SSL pretraining as well as linear eval.**

Prerequisite steps to test: 
1. Update ffcv to use version from ffcv-ssl repo. Run the following from outside the ffcv repo folder.
1. `pip uninstall ffcv`
2. `git clone https://github.com/facebookresearch/FFCV-SSL.git`
3. `pip install -e .`

**Summary of changes:**

TODO

**Test plan:**
1. Run BarlowTwins and SimCLR training with 2 augmentations and eval with 1 augmentation and check if previous numbers can be matched.
2. Run BarlowTwins and SimCLR training with 2 augmentations and precache and then eval with 1 augmentation and check if previous numbers can be matched.
3. Run BarlowTwins and SimCLR training with 2 augmentations and eval with 32 augmentations and check if linear eval accuracy numbers increase.
4. Run BarlowTwins and SimCLR training with 2, 4, 8, 16 augmentations for 10 epochs -- check if linear eval aaccuracies improve as number of augmentations increase.

This PR should close #31 

_EDIT 1:_ Checking multi-patch linear eval implementation for BarlowTwins pretraining.

| Num_augs | Precaching time | Linear eval accuracy |
|:--------:|:---------------:|:--------------------:|
|     1    |     35.21 s     |        83.75 %       |
|     8    |     180.65 s    |        84.52 %       |
|    16    |     368.04 s    |        85.02 %       |
|    32    |     765.69 s    |        85.00 %       |

### CODE REVIEW

**Review by kumarkrishna — APPROVED** [review]
generally LGTM, taking a closer look as I add SmaLR
